tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py

```diff
@@ -0,0 +1,148 @@
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph-fuse -n client.{name1} {fuse_mounting_dir_1} --client_fs cephfs",
+        )
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_1}{pin_dir}")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"setfattr -n ceph.dir.pin -v 1 {fuse_mounting_dir_1}{pin_dir}",
+        )
+        # checking is the dir is pinned
+        rank_0_mds = client1.exec_command(
+            sudo=True,
+            cmd="ceph fs status -f json-pretty | jq -r '.mdsmap[] | select(.rank == 0) | .name' ",
+        )
+        rank_0_mdss = rank_0_mds[0].split()
+        log.info(rank_0_mdss)
+        rank_1_mds = client1.exec_command(
+            sudo=True,
+            cmd="ceph fs status -f json-pretty | jq -r '.mdsmap[] | select(.rank == 1) | .name' ",
+        )
+        rank_1_mdss = rank_1_mds[0].split()
+        log.info(rank_1_mdss)
+        rank_0_request_prev = 0
+        rank_1_request_prev = 0
+        for mds in rank_0_mdss:
+            request = client1.exec_command(
+                sudo=True,
+                cmd=f"ceph tell mds.{mds} perf dump | jq -r '.mds | .request'",
+            )
+            request = int(request[0].strip())
+            rank_0_request_prev = rank_0_request_prev + int(request)
+        for mds in rank_1_mdss:
+            request = client1.exec_command(
+                sudo=True,
+                cmd=f"ceph tell mds.{mds} perf dump | jq -r '.mds | .request'",
+            )
+            request = request[0].strip()
+            rank_1_request_prev = rank_1_request_prev + int(request)
+        # running IOs
+        client1.exec_command(
+            sudo=True,
+            cmd=f"dd if=/dev/zero of={fuse_mounting_dir_1}{pin_dir}/{rand} bs=10240 count=1000 ",
+        )
+        time.sleep(30)
+        rank_0_request_after = 0
+        rank_1_request_after = 0
+        for mds in rank_0_mdss:
+            request = client1.exec_command(
+                sudo=True,
+                cmd=f"ceph tell mds.{mds} perf dump | jq -r '.mds | .request'",
+            )
+            request = int(request[0].strip())
+            rank_0_request_after = rank_0_request_after + int(request)
+        for mds in rank_1_mdss:
+            request = client1.exec_command(
+                sudo=True,
+                cmd=f"ceph tell mds.{mds} perf dump | jq -r '.mds | .request'",
+            )
+            request = int(request[0].strip())
+            rank_1_request_after = rank_1_request_after + int(request)
+        if rank_0_request_after > rank_0_request_prev:
+            log.info(f"rank_0_prev: {rank_0_request_prev}")
+            log.info(f"rank_0_after: {rank_0_request_after}")
+            log.info(f"rank_1_prev: {rank_1_request_prev}")
+            log.info(f"rank_1_after: {rank_1_request_after}")
+            raise CommandFailed("IOs are not pinned to rank 0")
```

Comment:
The test log uploaded is failing, http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-6K02VC/Active_Avtive_mds_race_condition_0.log, we need to find if the failure is expected or not,
if expected, handle the failure and not report an exception as it was expected.
If failure was not expected, and if its product issue, a BZ ID should be tagged , as other wise , a merge in this state, will cause a test failure in CI.

Reply:
I think BZ is not created so far, I am talking to VEnky regarding to this.. If it is a product issue, I will create a BZ and will add it.